public/app/plugins/datasource/azuremonitor/credentials.ts

```diff
@@ -166,7 +197,37 @@ export function updateCredentials(
           clientSecret: typeof credentials.clientSecret === 'symbol',
         },
       };
-
-      return options;
   }
+  if (instanceOfAzureCredential<AadCurrentUserCredentials>('currentuser', credentials)) {
+    const serviceCredentials = credentials.serviceCredentials;
+    let clientSecret: string | symbol | undefined;
+    if (instanceOfAzureCredential<AzureClientSecretCredentials>('clientsecret', serviceCredentials)) {
+      clientSecret = serviceCredentials.clientSecret;
+      // Do this to not expose the secret in unencrypted JSON data
+      delete serviceCredentials.clientSecret;
```

Comment:
Is this standard? this seems scary but I don't know how to get around it.

Reply:
We do this in a few data sources, it's a pretty standard way of removing a property from an object (also you're right there isn't really another way of doing it 😇).